packages/react-juce/src/components/Canvas.ts

```diff
@@ -1,66 +1,164 @@
+  // Rect functions
+  fillRect(x: number, y: number, width: number, height: number): void {
+    this._drawCommands.push(['fillRect', x, y, width, height]);
+  }
+
+  strokeRect(x: number, y: number, width: number, height: number): void {
+    this._drawCommands.push(['strokeRect', x, y, width, height]);
+  }
+
+  strokeRoundedRect(x: number, y: number, width: number, height: number, cornerSize: number): void {
+    this._drawCommands.push(['strokeRoundedRect', x, y, width, height, cornerSize]);
+  }
+
+  fillRoundedRect(x: number, y: number, width: number, height: number, cornerSize: number): void {
+    this._drawCommands.push(['fillRoundedRect', x, y, width, height, cornerSize]);
+  }
+
+  clearRect(x: number, y: number, width: number, height: number): void {
+    this._drawCommands.push(['clearRect', x, y, width, height]);
+  }
+
+  //================================================================================
+  // Path functions
+  //
+  // TODO: Should we split things out into CanvasRenderingContextPath
+  //       so things like close()/stroke() are more obvious when looking at API.
+  //       If you look at the ts type definitions for CanvasRenderingContext2D
+  //       CanvasRenderingContext2D is composed of other objects like CanvasRenderingContextPath
+  //       which contains all path methods. What is the best way to do this in JS and share
+  //       a drawCommands instance?
+  beginPath(): void {
+    this._drawCommands.push(['beginPath']);
+  }
+
+  lineTo(x: number, y: number): void {
+    this._drawCommands.push(['lineTo', x, y]);
+  }
+
+  moveTo(x: number, y: number): void {
+    this._drawCommands.push(['moveTo', x, y]);
+  }
+
+  arc(x: number, y: number, radius: number, startAngle: number, endAngle: number): void {
+    //TODO: Add support for optional antiClockWise?: boolean arg
+    this._drawCommands.push(['arc', x, y, radius, startAngle, endAngle]);
+  }
+
+  quadraticCurveTo(cpx: number, cpy: number, x: number, y: number): void {
+    this._drawCommands.push(['quadraticCurveTo', cpx, cpy, x, y]);
+  }
+
+  closePath(): void {
+    this._drawCommands.push(['close']);
+  }
+
+  stroke(): void {
+    this._drawCommands.push(['stroke']);
+  }
+
+  fill(): void {
+    this._drawCommands.push(['fill']);
+  }
+
+  //================================================================================
+  // Transform functions
+  rotate(angle: number): void {
+    this._drawCommands.push(['rotate', angle]);
+  }
+
+  translate(x: number, y: number): void {
+    this._drawCommands.push(['translate', x, y]);
+  }
+
+  setTransform(a: number, b: number, c: number, d: number, e: number, f: number): void {
+    this._drawCommands.push(['setTransform', a, b, c, d, e, f]);
+  }
+
+  resetTransform(): void {
+    this._drawCommands.push(['resetTransform']);
+  }
+
+  //================================================================================
+  // Image functions
+  //
+  //TODO: Add support for other drawImage overloads.
```

Comment:
I think maybe we can match the `source` prop API on `ImageView` here to support various image types

Reply:
Yeah awesome. This was actually just a comment pulled over from the original code. Could we take that as separate issue/PR ? I've deliberately avoided implementing any of the existing TODO's or additional API routines in this PR. Hoping to follow up with those next.

Reply:
Yep of course